File: src/1446_check_if_all_ones_are_at_least_length_k_places_away/CheckIfAllOnesAreAtLeastLengthKPlacesAway.java
public class CheckIfAllOnesAreAtLeastLengthKPlacesAway {
    public boolean kLengthApart(int[] nums, int k) {
        int previousId = -1;

        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == 1) {
                if (previousId != -1 && i - previousId - 1 < k)
                    return false;

                previousId = i;
            }
        }
        return true;
    }

    public static void main(String[] args) {
        CheckIfAllOnesAreAtLeastLengthKPlacesAway s = new CheckIfAllOnesAreAtLeastLengthKPlacesAway();
        int[] nums1 = { 1, 0, 0, 0, 1, 0, 0, 1 };
        int k1 = 2;
        System.out.println(s.kLengthApart(nums1, k1) + " true");
        int[] nums2 = { 1, 0, 0, 1, 0, 1 };
        int k2 = 2;
        System.out.println(s.kLengthApart(nums2, k2) + " false");
    }
}

File: src/1446_check_if_all_ones_are_at_least_length_k_places_away/check_if_all_ones_are_at_least_length_k_places_away.py
class Solution:
    def kLengthApart(self, nums, k):
        previousId = -1

        for i in range(len(nums)):
            if nums[i] == 1:
                if previousId != -1 and i - previousId - 1 < k:
                    return False
                previousId = i

        return True

s = Solution()
nums = [1, 0, 0, 0, 1, 0, 0, 1]
k = 2
print(s.kLengthApart(nums, k), True)
nums = [1, 0, 0, 1, 0, 1]
k = 2
print(s.kLengthApart(nums, k), False)
